src/test/scala/PSO/ParticleTopologySpec.scala

```diff
@@ -0,0 +1,239 @@
+      within(500.millis) {
+        ParticleTopology.formRing(particles)
+        //first connected to last and second
+        particles.head ! GetNeighbourhood
+        expectMsg(
+          CurrentNeighbourhood(Set(particles.takeRight(1).head, particles(1)))
+        )
+        //last connected to first and second last
+        particles.takeRight(1).head ! GetNeighbourhood
+        expectMsg(
+          CurrentNeighbourhood(Set(particles.head, particles.takeRight(2).head))
+        )
+        (1 until particles.length - 1).foreach{ x =>
+          particles(x) ! GetNeighbourhood
+          expectMsg(CurrentNeighbourhood(Set(particles(x - 1), particles(x + 1))))
+        }
+        //clear neighbourhoods for next test
+        particles.foreach(_ ! UpdateNeighbourhood(Set()))
+      }
+    }
+   
+    "be connected in a fully connected topology" in {
+      within(500.millis) {
+        val particleSet = particles.to[Set]
+        ParticleTopology.fullyConnected(particleSet)
+        //every particle connected to every other particle
+        particleSet.foreach{ p =>
+          p ! GetNeighbourhood
+          expectMsg(CurrentNeighbourhood(particleSet.diff(Set(p))))
+          //clear neighbourhoods for next test
+          p ! UpdateNeighbourhood(Set())
+        }
+      }
+    }
+
+    "be connected in a grid topology" in {
+      within(500.millis) {
+        ParticleTopology.formGrid(particles)
+        //top left corner
+        particles.head ! GetNeighbourhood
+        expectMsg(CurrentNeighbourhood(Set(particles(1), particles(3))))
+        //top side
+        particles(1) ! GetNeighbourhood
+        expectMsg(
+          CurrentNeighbourhood(Set(particles.head, particles(2), particles(4)))
+        )
+        //top right corner
+        particles(2) ! GetNeighbourhood
+        expectMsg(CurrentNeighbourhood(Set(particles(1), particles(5))))
+        //left side
+        particles(3) ! GetNeighbourhood
+        expectMsg(
+          CurrentNeighbourhood(Set(particles.head, particles(4), particles(6)))
+        )
+        //center
+        particles(4) ! GetNeighbourhood
+        expectMsg(
+          CurrentNeighbourhood(
+            Set(particles(1), particles(3), particles(5), particles(7))
+          )
+        )
+        //right side
+        particles(5) ! GetNeighbourhood
+        expectMsg(CurrentNeighbourhood(Set(particles(2), particles(4), particles(8))))
+        //bottom left corner
+        particles(6) ! GetNeighbourhood
+        expectMsg(CurrentNeighbourhood(Set(particles(3), particles(7))))
+        //bottom side
+        particles(7) ! GetNeighbourhood
+        expectMsg(CurrentNeighbourhood(Set(particles(4), particles(6), particles(8))))
+        //bottom right corner
+        particles(8) ! GetNeighbourhood
+        expectMsg(CurrentNeighbourhood(Set(particles(5), particles(7))))
+        //clear neighbourhoods for next test
+        particles.foreach(_ ! UpdateNeighbourhood(Set()))
```

Comment:
We're using BeforeAndAfter, so you should have nice container functions you can use to clear the topologies after each test